This can only be merged once the job is green. Here are the failures:
```
FAILED numpy/core/tests/test_mem_policy.py::test_new_policy - AssertionError:...
FAILED numpy/core/tests/test_umath.py::TestComplexFunctions::test_precisions_consistent
FAILED numpy/core/tests/test_umath.py::TestComplexFunctions::test_branch_cuts
FAILED numpy/core/tests/test_umath.py::TestComplexFunctions::test_branch_cuts_complex64
FAILED numpy/core/tests/test_umath.py::TestComplexFunctions::test_loss_of_precision[complex64]
FAILED numpy/core/tests/test_umath.py::TestComplexFunctions::test_loss_of_precision[complex128]
FAILED numpy/core/tests/test_umath.py::TestComplexFunctions::test_loss_of_precision[complex256]
FAILED numpy/linalg/tests/test_linalg.py::TestInv::test_sq_cases - AssertionE...
FAILED numpy/linalg/tests/test_linalg.py::TestInv::test_generalized_sq_cases
FAILED numpy/linalg/tests/test_linalg.py::TestCholesky::test_basic_property[float32-shape3]
FAILED numpy/linalg/tests/test_linalg.py::TestCholesky::test_basic_property[float64-shape3]
```

I'd say that the `TestComplexFunctions` are not very interesting, let's mark them as `xfail` for musl. The `test_mem_policy.py::test_new_policy` is also not a blocker, I suggest to xfail as well and open a separate issue for it. A bonus would be to include the Docker commands to reproduce the failure locally.

That leaves only the four linalg ones. I don't know what is going on there, but it looks important to check that soon. It looks similar to failures reported on macOS arm64 (https://github.com/conda-forge/numpy-feedstock/issues/287#issuecomment-1384881479), so very likely there's a real bug there that needs fixing soon. Let's also mark that xfail on musl and open a separate issue.

--- a/.github/workflows/linux_musl.yml
+++ b/.github/workflows/linux_musl.yml
@@ -1,0 +1,63 @@
+name: Test musllinux_x86_64
+
+on:
+  push:
+    branches:
+      - main
+      - maintenance/**
+  pull_request:
+    branches:
+      - main
+      - maintenance/**
+
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+
+permissions:
+  contents: read # to fetch code (actions/checkout)
+
+
+jobs:
+  musllinux_x86_64:
+    runs-on: ubuntu-latest
+    container:
+      # Use container used for building musllinux wheels
+      # it has git installed, all the pythons, etc
+      image: quay.io/pypa/musllinux_1_1_x86_64
+
+    steps:
+    - name: setup
+      run: |
+        apk update --quiet
+        apk add openblas-dev openblas
+
+        # using git commands to clone because versioneer doesn't work when
+        # actions/checkout is used for the clone step in a container
+
+        git config --global --add safe.directory $PWD 
+        
+        if [ $GITHUB_EVENT_NAME != pull_request ]; then
+            git clone --recursive --branch=$GITHUB_REF https://github.com/${GITHUB_REPOSITORY}.git $GITHUB_WORKSPACE
+            git reset --hard $GITHUB_SHA
+        else        
+            git clone --recursive https://github.com/${GITHUB_REPOSITORY}.git $GITHUB_WORKSPACE
+            git fetch origin $GITHUB_REF:my_ref_name
+            git checkout $GITHUB_BASE_REF
+            git -c user.email="you@example.com" merge --no-commit my_ref_name
+        fi
+
+        ln -s /usr/local/bin/python3.10 /usr/local/bin/python
+
+    - name: test musllinux_x86_64
+      run: |
+        python -m venv test_env
+        source test_env/bin/activate
+        python -m pip install -r test_requirements.txt
+
+        # calculate version information. Note that this doesn't work when
+        # actions/checkout is used for the cloning.
+        python setup.py version
+        python runtests.py --show-build-log --cpu-baseline=native --debug-info --mode=full -- -rsx -v

--- a/.github/workflows/linux_musl.yml
+++ b/.github/workflows/linux_musl.yml
@@ -60,4 +60,4 @@ jobs:
         # calculate version information. Note that this doesn't work when
         # actions/checkout is used for the cloning.
         python setup.py version
-        python runtests.py --show-build-log --cpu-baseline=native --debug-info --mode=full -- -rsx -v
+        python runtests.py --show-build-log --cpu-baseline=native --debug-info --mode=full -- -rsfx -v

--- a/tools/openblas_support.py
+++ b/tools/openblas_support.py
@@ -8,6 +8,7 @@ import shutil
 import tarfile
 import textwrap
 import zipfile
+from packaging.tags import sys_tags
 
 from tempfile import mkstemp, gettempdir
 from urllib.request import urlopen, Request
@@ -20,6 +21,7 @@ BASEURL = f'{BASE_LOC}/{OPENBLAS_LONG}/download'
 SUPPORTED_PLATFORMS = [
     'linux-aarch64',
     'linux-x86_64',
+    'musllinux-x86_64'
     'linux-i686',
     'linux-ppc64le',
     'linux-s390x',
@@ -55,10 +57,25 @@ def get_ilp64():
 
 def get_manylinux(arch):
     default = '2014'
-    ret = os.environ.get("MB_ML_VER", default)
+    ml_ver = os.environ.get("MB_ML_VER", default)
     # XXX For PEP 600 this can be a glibc version
-    assert ret in ('2010', '2014', '_2_24'), f'invalid MB_ML_VER {ret}'
-    return ret
+    assert ml_ver in ('1', '2010', '2014', '_2_24'), f'invalid MB_ML_VER {ml_ver}'
+    suffix = f'manylinux{ml_ver}_{arch}.tar.gz'
+    return suffix
+
+
+def get_musllinux(arch):
+    musl_ver = "1_1"
+    suffix = f'musllinux_{musl_ver}_{arch}.tar.gz'
+    return suffix
+
+
+def get_linux(arch):
+    tags = list(sys_tags())
+    if 'manylinux' in tags[0].platform:
+        return get_manylinux(arch)
+    elif 'musllinux' in tags[0].platform:
+        return get_musllinux(arch)
 
 
 def download_openblas(target, plat, ilp64):
@@ -70,8 +87,7 @@ def download_openblas(target, plat, ilp64):
                 '(KHTML, like Gecko) Chrome/41.0.2228.0 Safari/537.3')}
     suffix = None
     if osname == "linux":
-        ml_ver = get_manylinux(arch)
-        suffix = f'manylinux{ml_ver}_{arch}.tar.gz'
+        suffix = get_linux(arch)
         typ = 'tar.gz'
     elif plat == 'macosx-x86_64':
         suffix = 'macosx_10_9_x86_64-gf_c469a42.tar.gz'

